src/Site.js

```diff
@@ -231,119 +153,17 @@ function setExtension(filename, ext) {
  * Generate the site.json and an index.md file.
  *
  * @param rootPath
+ * @param templatePath
  */
-Site.initSite = function (rootPath) {
-  const boilerplatePath = path.join(rootPath, BOILERPLATE_FOLDER_NAME);
-  const configPath = path.join(rootPath, SITE_CONFIG_NAME);
-  const footerPath = path.join(rootPath, FOOTER_PATH);
-  const headerPath = path.join(rootPath, HEADER_PATH);
-  const headFolderPath = path.join(rootPath, HEAD_FOLDER_PATH);
-  const indexPath = path.join(rootPath, INDEX_MARKDOWN_FILE);
-  const siteNavPath = path.join(rootPath, SITE_NAV_PATH);
-  const siteLayoutPath = path.join(rootPath, LAYOUT_FOLDER_PATH);
-  const siteLayoutDefaultPath = path.join(siteLayoutPath, LAYOUT_DEFAULT_NAME);
-  const siteDefaultLayoutScriptsPath = path.join(siteLayoutDefaultPath, LAYOUT_SCRIPTS_PATH);
-  const sitePluginPath = path.join(rootPath, PROJECT_PLUGIN_FOLDER_NAME);
-  const userDefinedVariablesPath = path.join(rootPath, USER_VARIABLES_PATH);
-  // TODO: log the generate info
+Site.initSite = function (rootPath, templatePath = 'default') {
```

Comment:
The default value is not necessary since the caller in index.js already supplies the default value.